File: 0001. Two Sum.cpp
/*
Runtime: 8 ms, faster than 92.92% of C++ online submissions for Two Sum.
Memory Usage: 9.6 MB, less than 58.58% of C++ online submissions for Two Sum.
*/
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<pair<int,int> > v(nums.size());
        for(int i = 0; i < v.size(); ++i)
            v[i] = make_pair(nums[i], i);
        sort(v.begin(), v.end());
        
        int l = 0, r = nums.size()-1;
        
        while (l < r) {
            int s = v[l].first + v[r].first;
            if      (s < target) l++;
            else if (s > target) r--;
            else{
                vector<int> ret({v[l].second,v[r].second});
                sort(ret.begin(), ret.end());
                return ret;
            }
        }
        return vector<int>();
    }
};

File: 0001. Two Sum.py
'''
Runtime: 78 ms. Beats 79.76%.
Memory: 15.6 MB. Beats 9.31%.
'''
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        tuples = []
        for i in range(len(nums)):
            tuples.append([nums[i], i])
        
        tuples.sort()

        first = 0
        second = len(tuples) - 1

        while first < second:
            sum_ = tuples[first][0] + tuples[second][0]
            if sum_ == target:
                return [tuples[first][1], tuples[second][1]]
            elif sum_ > target:
                second -= 1
            else:
                first += 1



'''
Runtime: 3843 ms. Beats 18.73%.
Memory: 14.8 MB. Beats 95.49%.
'''
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        for i in range(len(nums) - 1):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]